File: sample/src/main/java/com/pierfrancescosoffritti/aytplayersample/examples/chromecastExample/utils/MediaRouteButtonUtils.java
package com.pierfrancescosoffritti.aytplayersample.examples.chromecastExample.utils;

import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.drawable.Drawable;
import android.support.v4.content.ContextCompat;
import android.support.v4.graphics.drawable.DrawableCompat;
import android.support.v7.app.MediaRouteButton;
import android.support.v7.view.ContextThemeWrapper;

import com.google.android.gms.cast.framework.CastButtonFactory;
import com.pierfrancescosoffritti.aytplayersample.R;
import com.pierfrancescosoffritti.aytplayersample.examples.chromecastExample.MediaRouteButtonContainer;

import javax.annotation.Nullable;

public class MediaRouteButtonUtils {
    static public MediaRouteButton initMediaRouteButton(Context context) {
        MediaRouteButton mediaRouteButton = new MediaRouteButton(context);
        CastButtonFactory.setUpMediaRouteButton(context, mediaRouteButton);

        return mediaRouteButton;
    }

    static public void addMediaRouteButtonToPlayerUI(
            MediaRouteButton mediaRouteButton, int tintColor,
            @Nullable MediaRouteButtonContainer disabledContainer, MediaRouteButtonContainer activatedContainer) {

        setMediaRouterButtonTint(mediaRouteButton, tintColor);

        if(disabledContainer != null)
        disabledContainer.removeMediaRouteButton(mediaRouteButton);

        if(mediaRouteButton.getParent() != null)
            return;

        activatedContainer.addMediaRouteButton(mediaRouteButton);
    }

    static private void setMediaRouterButtonTint(MediaRouteButton mediaRouterButton, int color) {
        ContextThemeWrapper castContext = new ContextThemeWrapper(mediaRouterButton.getContext(), R.style.Theme_MediaRouter);
        TypedArray styledAttributes = castContext.obtainStyledAttributes(null, android.support.v7.mediarouter.R.styleable.MediaRouteButton, android.support.v7.mediarouter.R.attr.mediaRouteButtonStyle, 0);
        Drawable drawable = styledAttributes.getDrawable(android.support.v7.mediarouter.R.styleable.MediaRouteButton_externalRouteEnabledDrawable);

        styledAttributes.recycle();
        DrawableCompat.setTint(drawable, ContextCompat.getColor(mediaRouterButton.getContext(), color));

        mediaRouterButton.setRemoteIndicatorDrawable(drawable);
    }
}

File: chromecast-sender-sample-app/src/main/java/com/pierfrancescosoffritti/cyplayersample/utils/MediaRouteButtonUtils.kt
package com.pierfrancescosoffritti.cyplayersample.utils

import android.content.Context
import androidx.core.content.ContextCompat
import androidx.core.graphics.drawable.DrawableCompat
import androidx.mediarouter.app.MediaRouteButton
import androidx.mediarouter.R
import android.view.ContextThemeWrapper
import com.google.android.gms.cast.framework.CastButtonFactory
import com.pierfrancescosoffritti.cyplayersample.examples.localPlayerExample.MediaRouteButtonContainer

object MediaRouteButtonUtils {

    fun initMediaRouteButton(context: Context) : MediaRouteButton {
        val mediaRouteButton = MediaRouteButton(context)
        CastButtonFactory.setUpMediaRouteButton(context, mediaRouteButton)

        return mediaRouteButton
    }

    fun initMediaRouteButton(mediaRouteButton: MediaRouteButton) : MediaRouteButton {
        CastButtonFactory.setUpMediaRouteButton(mediaRouteButton.context, mediaRouteButton)
        return mediaRouteButton
    }

    fun addMediaRouteButtonToPlayerUi(
            mediaRouteButton: MediaRouteButton, tintColor: Int,
            disabledContainer: MediaRouteButtonContainer?, activatedContainer: MediaRouteButtonContainer) {

        setMediaRouterButtonTint(mediaRouteButton, tintColor)

        disabledContainer?.removeMediaRouteButton(mediaRouteButton)
        if(mediaRouteButton.parent != null) return
        activatedContainer.addMediaRouteButton(mediaRouteButton)
    }

    private fun setMediaRouterButtonTint(mediaRouterButton: MediaRouteButton, color: Int) {
        val castContext = ContextThemeWrapper(mediaRouterButton.context, R.style.Theme_MediaRouter)
        val styledAttributes = castContext.obtainStyledAttributes(null, R.styleable.MediaRouteButton, R.attr.mediaRouteButtonStyle, 0)
        val drawable = styledAttributes.getDrawable(R.styleable.MediaRouteButton_externalRouteEnabledDrawable)

        styledAttributes.recycle()
        DrawableCompat.setTint(drawable, ContextCompat.getColor(mediaRouterButton.context, color))

        mediaRouterButton.setRemoteIndicatorDrawable(drawable)
    }
}